external/sql/storm-sql-runtime/src/test/org/apache/storm/sql/TestUtils.java

```diff
@@ -60,6 +62,25 @@ public static String result(String accumulator) {
   }
 
 
+  public static class TopN {
+    public static PriorityQueue<Integer> init() {
+      return new PriorityQueue<>();
+    }
+    public static PriorityQueue<Integer> add(PriorityQueue<Integer> accumulator, Integer n, Integer val) {
+      if (accumulator.size() >= n) {
```

Comment:
Logic need to be changed to add val first, and remove the lowest `if accumulator.size() > n` because val could be lower than lowest value of priority queue.
Suppose the input `3, 2, 1`, accumulator will be changed to 3 -> 2, 3 -> 1, 3 (should be 2, 3).

We could even do nothing when `(n > 0 && accumulator.size() >= n && accumulator.peek() >= val)`.